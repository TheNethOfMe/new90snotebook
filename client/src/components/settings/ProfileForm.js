import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";

import { userChangeTheme } from "../../actions/authActions";

import TextFieldGroup from "../common/TextFieldGroup";
import DropDownGroup from "../common/DropDownGroup";

export class ProfileForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      firstName: props.profile ? props.profile.firstName : "",
      lastName: props.profile ? props.profile.lastName : "",
      nickName:
        props.profile && props.profile.nickName ? props.profile.nickName : "",
      theme: props.theme,
      errors: props.errors
    };
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.errors) {
      this.setState({ errors: nextProps.errors });
    }
  }
  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  pickTheme = e => {
    const theme = e.target.value;
    this.setState({ theme });
    this.props.userChangeTheme({ theme });
  };
  onSubmit = e => {
    e.preventDefault();
    this.props.handleSubmit({
      firstName: this.state.firstName,
      lastName: this.state.lastName,
      nickName: this.state.nickName,
      bio: this.state.bio
    });
  };
  render() {
    const { errors } = this.state;
    const themeOptions = [
      {
        val: "paper-cup",
        display: "Paper Cup"
      },
      {
        val: "food-court",
        display: "Food Court"
      },
      {
        val: "frankly-lisa",
        display: "Frankly Lisa"
      },
      {
        val: "old-mac",
        display: "Old Mac"
      },
      {
        val: "tropical",
        display: "Tropical"
      }
    ];
    return (
      <div>
        <form onSubmit={this.onSubmit}>
          <TextFieldGroup
            name="firstName"
            placeholder="firstName"
            value={this.state.firstName}
            label="First Name"
            error={errors.firstName}
            onChange={this.onChange}
          />
          <TextFieldGroup
            name="lastName"
            placeholder="lastName"
            value={this.state.lastName}
            label="Last Name"
            error={errors.lastName}
            onChange={this.onChange}
          />
          <TextFieldGroup
            name="nickName"
            placeholder="nickName"
            value={this.state.nickName}
            label="Nick Name"
            error={errors.nickName}
            onChange={this.onChange}
          />
          <div className="input">
            <label className="input__label" htmlFor="bio">
              Tell us about yourself.
            </label>
            <textarea
              className="input__field"
              name="bio"
              onChange={this.onChange}
              value={this.state.bio}
            />
          </div>
          <DropDownGroup
            name="theme"
            label="Choose a theme"
            onChange={this.pickTheme}
            options={themeOptions}
            selected={this.state.theme}
          />
          <input className="input__submit" type="submit" value="Finished!" />
        </form>
      </div>
    );
  }
}

ProfileForm.propTypes = {
  profile: PropTypes.object,
  errors: PropTypes.object.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  userChangeTheme: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  theme: state.auth.user.theme
});

export default connect(
  mapStateToProps,
  { userChangeTheme }
)(ProfileForm);
