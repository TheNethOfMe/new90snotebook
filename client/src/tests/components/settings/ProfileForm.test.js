import React from "react";
import { shallow } from "enzyme";
import ProfileForm from "../../../components/settings/ProfileForm";
import { fakeUserProfile } from "../../dummyData";

let wrapper, handleSubmit, errors;

beforeEach(() => {
  handleSubmit = jest.fn();
  errors = {};
});

test("should render the profile form", () => {
  wrapper = shallow(
    <ProfileForm handleSubmit={handleSubmit} profile={null} errors={errors} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render profile form with values in fields if a profile is passed in", () => {
  wrapper = shallow(
    <ProfileForm
      handleSubmit={handleSubmit}
      profile={fakeUserProfile}
      errors={errors}
    />
  );
  const checkFirstName = wrapper
    .find("TextFieldGroup")
    .at(0)
    .props().value;
  const checkLastName = wrapper
    .find("TextFieldGroup")
    .at(1)
    .props().value;
  expect(checkFirstName).toEqual("Sparky");
  expect(checkLastName).toEqual("McSparkerton");
});
