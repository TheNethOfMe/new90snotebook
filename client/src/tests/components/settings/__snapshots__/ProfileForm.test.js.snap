// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the profile form 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileForm
    errors={Object {}}
    handleSubmit={[MockFunction]}
    profile={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <TextFieldGroup
          disabled={false}
          label="First Name"
          name="firstName"
          onChange={[Function]}
          placeholder="firstName"
          type="text"
          value=""
        />
        <TextFieldGroup
          disabled={false}
          label="Last Name"
          name="lastName"
          onChange={[Function]}
          placeholder="lastName"
          type="text"
          value=""
        />
        <TextFieldGroup
          disabled={false}
          label="Nick Name"
          name="nickName"
          onChange={[Function]}
          placeholder="nickName"
          type="text"
          value=""
        />
        <DropDownGroup
          disabled={false}
          info="Hidden profiles can not be found or friended by anyone else."
          label="Do you want to be searchable?"
          name="searchableProfile"
          onChange={[Function]}
          options={
            Array [
              Object {
                "display": "Make my profile hidden",
                "val": false,
              },
              Object {
                "display": "Make my profile searchable",
                "val": true,
              },
            ]
          }
          type="text"
        />
        <DropDownGroup
          disabled={false}
          label="Choose a theme"
          name="theme"
          onChange={[Function]}
          options={
            Array [
              Object {
                "display": "Paper Cup",
                "val": "paper-cup",
              },
              Object {
                "display": "Tropical",
                "val": "tropical",
              },
              Object {
                "display": "Frankly Lisa",
                "val": "frankly-lisa",
              },
            ]
          }
          type="text"
        />
        <input
          className="input__submit"
          type="submit"
          value="Finished!"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TextFieldGroup
            disabled={false}
            label="First Name"
            name="firstName"
            onChange={[Function]}
            placeholder="firstName"
            type="text"
            value=""
          />,
          <TextFieldGroup
            disabled={false}
            label="Last Name"
            name="lastName"
            onChange={[Function]}
            placeholder="lastName"
            type="text"
            value=""
          />,
          <TextFieldGroup
            disabled={false}
            label="Nick Name"
            name="nickName"
            onChange={[Function]}
            placeholder="nickName"
            type="text"
            value=""
          />,
          <DropDownGroup
            disabled={false}
            info="Hidden profiles can not be found or friended by anyone else."
            label="Do you want to be searchable?"
            name="searchableProfile"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "display": "Make my profile hidden",
                  "val": false,
                },
                Object {
                  "display": "Make my profile searchable",
                  "val": true,
                },
              ]
            }
            type="text"
          />,
          <DropDownGroup
            disabled={false}
            label="Choose a theme"
            name="theme"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "display": "Paper Cup",
                  "val": "paper-cup",
                },
                Object {
                  "display": "Tropical",
                  "val": "tropical",
                },
                Object {
                  "display": "Frankly Lisa",
                  "val": "frankly-lisa",
                },
              ]
            }
            type="text"
          />,
          <input
            className="input__submit"
            type="submit"
            value="Finished!"
          />,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "error": undefined,
            "label": "First Name",
            "name": "firstName",
            "onChange": [Function],
            "placeholder": "firstName",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "error": undefined,
            "label": "Last Name",
            "name": "lastName",
            "onChange": [Function],
            "placeholder": "lastName",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "error": undefined,
            "label": "Nick Name",
            "name": "nickName",
            "onChange": [Function],
            "placeholder": "nickName",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "info": "Hidden profiles can not be found or friended by anyone else.",
            "label": "Do you want to be searchable?",
            "name": "searchableProfile",
            "onChange": [Function],
            "options": Array [
              Object {
                "display": "Make my profile hidden",
                "val": false,
              },
              Object {
                "display": "Make my profile searchable",
                "val": true,
              },
            ],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "label": "Choose a theme",
            "name": "theme",
            "onChange": [Function],
            "options": Array [
              Object {
                "display": "Paper Cup",
                "val": "paper-cup",
              },
              Object {
                "display": "Tropical",
                "val": "tropical",
              },
              Object {
                "display": "Frankly Lisa",
                "val": "frankly-lisa",
              },
            ],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input__submit",
            "type": "submit",
            "value": "Finished!",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <TextFieldGroup
            disabled={false}
            label="First Name"
            name="firstName"
            onChange={[Function]}
            placeholder="firstName"
            type="text"
            value=""
          />
          <TextFieldGroup
            disabled={false}
            label="Last Name"
            name="lastName"
            onChange={[Function]}
            placeholder="lastName"
            type="text"
            value=""
          />
          <TextFieldGroup
            disabled={false}
            label="Nick Name"
            name="nickName"
            onChange={[Function]}
            placeholder="nickName"
            type="text"
            value=""
          />
          <DropDownGroup
            disabled={false}
            info="Hidden profiles can not be found or friended by anyone else."
            label="Do you want to be searchable?"
            name="searchableProfile"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "display": "Make my profile hidden",
                  "val": false,
                },
                Object {
                  "display": "Make my profile searchable",
                  "val": true,
                },
              ]
            }
            type="text"
          />
          <DropDownGroup
            disabled={false}
            label="Choose a theme"
            name="theme"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "display": "Paper Cup",
                  "val": "paper-cup",
                },
                Object {
                  "display": "Tropical",
                  "val": "tropical",
                },
                Object {
                  "display": "Frankly Lisa",
                  "val": "frankly-lisa",
                },
              ]
            }
            type="text"
          />
          <input
            className="input__submit"
            type="submit"
            value="Finished!"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TextFieldGroup
              disabled={false}
              label="First Name"
              name="firstName"
              onChange={[Function]}
              placeholder="firstName"
              type="text"
              value=""
            />,
            <TextFieldGroup
              disabled={false}
              label="Last Name"
              name="lastName"
              onChange={[Function]}
              placeholder="lastName"
              type="text"
              value=""
            />,
            <TextFieldGroup
              disabled={false}
              label="Nick Name"
              name="nickName"
              onChange={[Function]}
              placeholder="nickName"
              type="text"
              value=""
            />,
            <DropDownGroup
              disabled={false}
              info="Hidden profiles can not be found or friended by anyone else."
              label="Do you want to be searchable?"
              name="searchableProfile"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "display": "Make my profile hidden",
                    "val": false,
                  },
                  Object {
                    "display": "Make my profile searchable",
                    "val": true,
                  },
                ]
              }
              type="text"
            />,
            <DropDownGroup
              disabled={false}
              label="Choose a theme"
              name="theme"
              onChange={[Function]}
              options={
                Array [
                  Object {
                    "display": "Paper Cup",
                    "val": "paper-cup",
                  },
                  Object {
                    "display": "Tropical",
                    "val": "tropical",
                  },
                  Object {
                    "display": "Frankly Lisa",
                    "val": "frankly-lisa",
                  },
                ]
              }
              type="text"
            />,
            <input
              className="input__submit"
              type="submit"
              value="Finished!"
            />,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "error": undefined,
              "label": "First Name",
              "name": "firstName",
              "onChange": [Function],
              "placeholder": "firstName",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "error": undefined,
              "label": "Last Name",
              "name": "lastName",
              "onChange": [Function],
              "placeholder": "lastName",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "error": undefined,
              "label": "Nick Name",
              "name": "nickName",
              "onChange": [Function],
              "placeholder": "nickName",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "info": "Hidden profiles can not be found or friended by anyone else.",
              "label": "Do you want to be searchable?",
              "name": "searchableProfile",
              "onChange": [Function],
              "options": Array [
                Object {
                  "display": "Make my profile hidden",
                  "val": false,
                },
                Object {
                  "display": "Make my profile searchable",
                  "val": true,
                },
              ],
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "label": "Choose a theme",
              "name": "theme",
              "onChange": [Function],
              "options": Array [
                Object {
                  "display": "Paper Cup",
                  "val": "paper-cup",
                },
                Object {
                  "display": "Tropical",
                  "val": "tropical",
                },
                Object {
                  "display": "Frankly Lisa",
                  "val": "frankly-lisa",
                },
              ],
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input__submit",
              "type": "submit",
              "value": "Finished!",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
